src/codegen.cpp

```diff
@@ -709,10 +709,11 @@ typedef struct {
 
 typedef struct {
     size_t len;
+    // variable length
     struct {
         int64_t isref;
         Function *f;
-    } data[];
+    } data[1];
```

Comment:
`std::vector< std::pair< int64_t, Function* > >` would also work. it's expected to be a very small set, so actually it's best not to use `unordered_map`, which may significantly over-allocate.